sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java

```diff
@@ -207,6 +257,36 @@ private PartitionWriter(TopicPartition tp, long firstOffset) throws IOException
               firstOffset,
               GZIPChunkThreshold,
               format.init(tp.topic(), tp.partition(), firstOffset));
+      lastFlush = Instant.now().toEpochMilli();
+    }
+
+    public BlockGZIPFileWriter getWriter() {
+      return writer;
+    }
+
+    public boolean shouldBeDone() {
+      long timeSinceLastFlush = Instant.now().toEpochMilli() - lastFlush;
+      boolean doPeriodicFlush = timeSinceLastFlush >= flushIntervalMs;
+      if (doPeriodicFlush) {
+        log.info("{} performing a periodic flush on {}", name(), tp);
+      }
+
+      boolean doFileSizeFlush = writer.getTotalCompressedSize() > GZIPFileThreshold;
+      log.info(
```

Comment:
The log statements should probably be DEBUG level as well. The `performing periodic flush` perhaps we could keep as INFO as it would only be on the `12h` interval.

Reply:
It's either both size and time flush messages are `info` level or `debug` as everything else. I'll change it to `debug` to be consistent.